### DISCUSSION TITLE
sigil_z

### DISCUSSION THREAD

**Discussion by NickNeck** [doorgan/sourceror#50] [Proposals]
Hi @doorgan 

I have a proposal for a sigil.
```elixir
defmodule Sourceror.Sigils do
  defmacro sigil_z({:<<>>, _, [string]}, []) do
    ast = Code.string_to_quoted!(string, file: __CALLER__.file, line: __CALLER__.line)
    {ast, {:_zipper_meta, [if_undefined: :apply], Elixir}}
  end
end
```

That `~z` could be used while traversing with a `Zipper`.
```elixir
  def traverse_fun(~z/{:|>, _meta, args}/ = zipper) do
    ...  
  end
```
instead of
```elixir
  def traverse_fun({{:|>, _meta, args}, _zipper_meta} = zipper) do
    ...  
  end
```
Advantage
  * focus on the ast
  
Disadvantage
  * no highlighting for the ast in `~z`
  * the function definition becomes more complex

What do you think?

**Comment by doorgan** [comment]
Hey there! Thanks for the suggestion!

I see the appeal to it, but I'm not sure the sigil justifies the tradeoff considering it saves just a few characters, and that we're replacing the zipper metadata matching with the sigil invocation.

I'll give it some more thought and get back to you :)

**Reply by NickNeck** [reply]
Hello, I completely agree with you. I am also not sure if this is a good feature. The problem is that any developer will understand the matching with a zipper-tuple, but the z-sigil in a definition could be weird. On the other hand the z-sigil makes things a little bit more readable. For now I tend to leave it out. But I wanted to hear some other opinions on this.
